### PR TITLE
Add missing unit tests

### PR DESCRIPTION
**Pull request by Manuri**
## Purpose
$Subject
1. Adds tests for SQL connector init failure
2. Add more negative tests for proxy table add operation
3. Correct a condition by replacing a null check with an empty check

### CODE REVIEW

File: tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/connectors/sql/SQLConnectorInitTest.java [hunk 1]
```diff
@@ -113,6 +114,13 @@ public void testPropertiesGetUsedOnlyIfDataSourceGiven() {
         Assert.assertEquals(returns[0].stringValue(), expected);
     }
 
+    @Test(expectedExceptions = BLangRuntimeException.class,
+          expectedExceptionsMessageRegExp = ".*error in sql connector configuration:Failed to initialize pool: "
+                  + "Database does not exists: ./target/tempdb/NON_EXISTING_DB.*")
+    public void testConnectionFailure() {
+        BRunUtil.invokeFunction(result, "testConnectionFailure");
+        Assert.fail("Expected exception should have been thrown by this point");
```
[the remark this answers is not in the document]
Comment (Manuri) [reply tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/connectors/sql/SQLConnectorInitTest.java]: Yes, this is not really necessary.

File: tests/ballerina-unit-test/src/test/resources/test-src/connectors/sql/sql_mirror_table_test.bal [hunk 2]
```diff
@@ -118,7 +125,26 @@ function testAddToMirrorTable(string jdbcUrl, string userName, string password)
...
+
+    table dt = check testDB->getProxyTable("employeeAdd", Employee);
+
+    Employee2 e = { id: 2, name: "Devni", address: "Sri Lanka", age: 24 };
+
+    var result = dt.add(e);
+
+    testDB.stop();
+
+    return result;
```
Comment (anupama-pathirage) [inline tests/ballerina-unit-test/src/test/resources/test-src/connectors/sql/sql_mirror_table_test.bal]: Shall we remove unnecessary new lines between code lines?

File: stdlib/database/h2/src/main/java/org/ballerinalang/database/h2/CreateClient.java [hunk 3]
```diff
@@ -52,7 +52,7 @@ public void execute(Context context) {
         Struct clientEndpointConfig = BLangConnectorSPIUtil.toStruct(configBStruct);
         Map<String, Value> dbOptions = clientEndpointConfig.getMapField(Constants.EndpointConfig.DB_OPTIONS);
         String urlOptions = "";
-        if (dbOptions != null) {
+        if (!dbOptions.isEmpty()) {
```
Comment (anupama-pathirage) [inline stdlib/database/h2/src/main/java/org/ballerinalang/database/h2/CreateClient.java]: Can't this become null ? 
[1] https://github.com/ballerina-platform/ballerina-lang/blob/master/bvm/ballerina-core/src/main/java/org/ballerinalang/connector/impl/StructImpl.java#L129

File: tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/connectors/sql/SQLActionsTest.java [hunk 4]
```diff
@@ -763,6 +763,25 @@ public void testAddToMirrorTableNegative() throws Exception {
         Assert.assertTrue(returns[0].stringValue().contains(errorMessage));
     }
 
+    @Test(groups = CONNECTOR_TEST, description = "Test failure scenario in adding data to mirrored table")
+    public void testAddToMirrorTableInvalidRecord() throws Exception {
```
Comment (anupama-pathirage) [inline tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/connectors/sql/SQLActionsTest.java]: Shall we reword test description and method name use proxy table

Comment (Manuri) [reply tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/connectors/sql/SQLActionsTest.java]: +1. I'll do this collectively for all places in a separate PR.

File: stdlib/database/h2/src/main/java/org/ballerinalang/database/h2/CreateClient.java (hunk 3 above)
Comment (Manuri) [reply stdlib/database/h2/src/main/java/org/ballerinalang/database/h2/CreateClient.java]: Even if we don't pass any dbOptions, an empty map will be present. I have checked at the Line 129 but seems an empty map is passed there too.

**Pull request merged**